hil-test/tests/spi_full_duplex.rs

```diff
@@ -0,0 +1,105 @@
+//! SPI Full Duplex Test
+//!
+//! Folowing pins are used:
+//! SCLK    GPIO0
+//! MISO    GPIO2
+//! MOSI    GPIO4
+//! CS      GPIO5
+//!
+//! Connect MISO (GPIO2) and MOSI (GPIO4) pins.
+
+#![no_std]
+#![no_main]
+
+use embedded_hal_1::spi::SpiBus;
+use hil_test::esp_hal::{
+    clock::ClockControl,
+    gpio::IO,
+    peripherals::Peripherals,
+    prelude::*,
+    spi::{master::Spi, FullDuplexMode, SpiMode},
+};
+
+struct Context {
+    spi: Spi<'static, esp_hal::peripherals::SPI2, FullDuplexMode>,
+}
+
+impl Context {
+    pub fn init() -> Self {
+        let peripherals = Peripherals::take();
+        let system = peripherals.SYSTEM.split();
+        let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+
+        let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
+        let sclk = io.pins.gpio0;
+        let miso = io.pins.gpio2;
+        let mosi = io.pins.gpio4;
+        let cs = io.pins.gpio5;
+
+        let spi = Spi::new(peripherals.SPI2, 1000u32.kHz(), SpiMode::Mode0, &clocks).with_pins(
+            Some(sclk),
+            Some(mosi),
+            Some(miso),
+            Some(cs),
+        );
+
+        Context { spi }
+    }
+}
+
+#[embedded_test::tests]
+mod tests {
+    use defmt::assert_eq;
+
+    use super::*;
+
+    #[init]
+    fn init() -> Context {
+        Context::init()
+    }
+
+    #[test]
+    fn test_symestric_transfer(mut ctx: Context) {
+        let write = [0xde, 0xad, 0xbe, 0xef];
+        let mut read: [u8; 4] = [0x00u8; 4];
+        SpiBus::transfer(&mut ctx.spi, &mut read[..], &write[..])
```

Comment:
We should be able to just use the `transfer` method now the traits aren't included in the prelude (the PR might need a rebase).